Protect isenumclass predicate against non-class arguments

bryevdv opened the pull request:
Subject: Protect isenumclass predicate against non-class arguments

### Feature or Bugfix
- Bugfix

### Purpose

Not being a class is a reasonable way a thing might not be an enum class,
but without an explicit check, isenumclass raises an exception in this case.

### Detail

fixes: #3731

### Relates
- <URL or Ticket>

tk0miya commented:
LGTM. Could you rebase this to stable branch?
Then I will merge this into next 1.5.x release.
Thanks,

tk0miya commented:
@shimizukawa do you have any comments?

shimizukawa reviewed (approved):
LGTM!

bryevdv committed:
Protect isenumclass predicate against non-class arguments

fixes: #3731

Not being a class is another way a thing might not be an enum class,
but without an explicit check, isenumclass raises an exception in this
case.

bryevdv commented:
> Could you rebase this to stable branch?

I got  a conflict when I did that:

```
~/work/sphinx stable
❯ go bryevdv/3731_protect_isenumclass
Switched to branch 'bryevdv/3731_protect_isenumclass'
Your branch is up-to-date with 'origin/bryevdv/3731_protect_isenumclass'.
~/work/sphinx bryevdv/3731_protect_isenumclass
❯ git rebase stable
First, rewinding head to replay your work on top of it...
Applying: Allow setting 'rel' and 'title' attributes for stylesheets.
Applying: Fix two statements in one line
Applying: Changed 'rel' to 'alternate'
Applying: Test for alternate stylesheets
Using index info to reconstruct a base tree...
M	tests/test_build_html.py
.git/rebase-apply/patch:13: new blank line at EOF.
+
.git/rebase-apply/patch:31: new blank line at EOF.
+
warning: 2 lines add whitespace errors.
Falling back to patching base and 3-way merge...
Auto-merging tests/test_build_html.py
CONFLICT (content): Merge conflict in tests/test_build_html.py
error: Failed to merge in the changes.
Patch failed at 0004 Test for alternate stylesheets
The copy of the patch that failed is found in: .git/rebase-apply/patch
```

So, I stashed changes, applied to stable, blew-away and re-branched, force pushed new copy of the branch. I also changed the base branch in the UI above. LMK if that's not what you needed.

tk0miya commented:
Thank you for updating! Merged!
